src/games/strategy/triplea/ai/proAI/data/ProTerritoryManager.java

```diff
@@ -1,31 +1,1144 @@
 package games.strategy.triplea.ai.proAI.data;
 
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.PlayerID;
+import games.strategy.engine.data.Route;
 import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.TripleAUnit;
+import games.strategy.triplea.ai.proAI.ProData;
+import games.strategy.triplea.ai.proAI.logging.ProLogger;
+import games.strategy.triplea.ai.proAI.util.ProBattleUtils;
+import games.strategy.triplea.ai.proAI.util.ProMatches;
+import games.strategy.triplea.ai.proAI.util.ProTransportUtils;
+import games.strategy.triplea.ai.proAI.util.ProUtils;
+import games.strategy.triplea.attatchments.TerritoryAttachment;
+import games.strategy.triplea.attatchments.UnitAttachment;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.MoveValidator;
+import games.strategy.triplea.delegate.TerritoryEffectHelper;
+import games.strategy.triplea.delegate.TransportTracker;
+import games.strategy.util.CompositeMatchAnd;
+import games.strategy.util.CompositeMatchOr;
+import games.strategy.util.Match;
 
 import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
 
+/**
+ * Manages info about territories.
+ */
 public class ProTerritoryManager {
 
-  final Map<Territory, ProTerritory> territoryMap;
+  private final PlayerID player;
 
-  public ProTerritoryManager(final Map<Territory, ProTerritory> territoryMap) {
-    this.territoryMap = territoryMap;
+  private ProMyMoveOptions attackOptions;
```

Comment:
Can't as currently designed since you'll often recalculate many of the maps as you gain more info through the phase. The best example is determining enemy attack options. When you first start the turn you check and find all enemy attack options but then you move some units to either attack or block enemy attacks so you then recalculate them.